app/v2/models/Binders.scala

```diff
@@ -21,11 +21,10 @@ import play.api.mvc.PathBindable
 object Binders {
 
   implicit val grnBinder: PathBindable[GuaranteeReferenceNumber] = new PathBindable[GuaranteeReferenceNumber] {
-    private val grnPattern = """[0-9]{2}([A-Z]{2})[A-Z0-9]{12}[0-9]([A-Z][0-9]{6})?""".r
+    val grnPattern = """[0-9]{2}(GB|XI)[A-Z0-9]{12}[0-9]([A-Z][0-9]{6})?""".r
```

Comment:
This was intended so we could give an appropriate error message below. Nor does this fix the actual bug -- I think this regex will still match a GRN that is too long (which was the actual bug).

Reply:
I have tested with various conditions. if you are passing more than valid length it will return 400 with **The guarantee reference number is not in the correct format.** and it is not going to pass with 200.

Reply:
Interesting, @kiranhaleyur reported otherwise, hence the issue that was made. You're right, I did just create another test to check this and it looks like it worked fine.

This change should still be reverted for the reasons I gave before.

Reply:
Really Interesting!
Correct me if my understanding is not what you guys are discussing,  
I am passing GRN - 32GBNS1DZBLXRXGB1N2802441222G in my test which is definitely exceeded the length, the error code I am getting is 400, but the response message getting is "**bad request, cause: REDACTED**"
few more grn tested: getting same error message
32GBNS1DZBLXRXGB1N280299546546
32GBNS1DZBLXRXGB1N280299546546G43433
34GB33
12345

@sonisn1988 : can you pass the GRN you have tested, I will try from my end. 
But looks like two things are happening here which needs to be figured out that looks worried.

Reply:
Right. There's been crossed wires somewhere with that ticket.

When we talked about that, I was under the impression that giving a GRN that is too long but a subsection of it being valid was returning a 200 - so something like

`32GBNS1DZBLXRXGB1N2802441222G`

was returning 

```json
{
    "_links": {
        "self": {
            "href": "/customs/guarantees/32GBNS1DZBLXRXGB1N2802441222G/balance"
        }
    },
    "balance": 1234.56
}
```

That's not the case.

The error message problems are one and the same bug, so will be fixed in the same way.

I'm content now. Sorry for any confusion there.